xdr3/decode.go

```diff
@@ -509,6 +511,10 @@ func (d *Decoder) decodeArray(v reflect.Value, ignoreOpaque bool, maxSize int, m
 	// existing slice does not have enough capacity.
 	sliceLen := int(dataLen)
 	if v.Cap() < sliceLen {
+		growth := (sliceLen - v.Cap()) * int(v.Type().Size())
```

Comment:
A little concerned about the conversion assuming uintptr is the same bitsize as int, although in practice I think that's usually the case it's not guaranteed by the language. The safest thing to do would be to do the conversion then check for overflow before using the value.